.github/workflows/main.yml

```diff
@@ -50,3 +50,27 @@ jobs:
           provider: microk8s
       - name: Run integration tests
         run: tox -e integration
+      - name: Archive Tested Charm
+        uses: actions/upload-artifact@v3
+        if: ${{ github.event_name == 'push' }}
+        with:
+          name: tested-charm
+          path: .tox/**/tls-certificates-operator_ubuntu-20.04-amd64.charm
+          retention-days: 5
+
+  publish-charm:
+    name: Publish Charm
+    needs: integration-test
+    runs-on: ubuntu-20.04
+    if: ${{ github.event_name == 'push' }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Fetch Tested Charm
+        uses: actions/download-artifact@v3
+        with:
+          name: tested-charm
+      - name: Upload charm to Charmhub edge channel
+        env:
+          CHARMCRAFT_AUTH: "${{ secrets.CHARMHUB_TOKEN }}"
+        run: charmcraft upload ./tls-certificates-operator_ubuntu-20.04-amd64.charm --release edge
```

Comment:
Wouldn't this on pull request creation?

Reply:
No, there is an `if` for the whole job limiting this to `push`, and the whole workflow is limited to the main branch (meaning pushes to the main branch, or PRs targeting the main branch)